File: day1/cpp/main.cpp
#include <iostream>
#include <unordered_map>
#include <string>
using namespace std;
class Solution {
    public:
        int firstUniqueChar(string s) {
            unordered_map<char, int> charCount;
            for (char c : s) {
                charCount[c]++;
            }
            for (int i = 0; i < s.size(); i++) {
                if (charCount[s[i]] == 1) {
                    return i;
                    }
            };
            return -1;
        };
};

File: day1/typescript/main.ts
export const firstUniqueChar = (s: string): number => {
    const charCount: { [key: string]: number } = {};
    
    for (const char of s) {
        charCount[char] = (charCount[char] || 0) + 1;
    }
    
    for (let i = 0; i < s.length; i++) {
        if (charCount[s[i]] === 1) {
        return i;
        }
    }
    
    return -1;
    }